Apps/WebClient/src/ClientApp/src/components/timeline/entryCard/laboratory.vue

```diff
@@ -67,15 +67,43 @@ export default class LaboratoryTimelineComponent extends Vue {
         this.laboratoryService
             .getReportDocument(this.entry.id, this.user.hdid)
             .then((result) => {
-                const link = document.createElement("a");
                 let dateString = this.entry.displayDate.format(
                     "YYYY_MM_DD-HH_mm"
                 );
                 let report: LaboratoryReport = result.resourcePayload;
-                link.href = `data:${report.mediaType};${report.encoding},${report.data}`;
-                link.download = `COVID_Result_${dateString}.pdf`;
-                link.click();
-                URL.revokeObjectURL(link.href);
+                fetch(
+                    `data:${report.mediaType};${report.encoding},${report.data}`
+                )
+                    .then((res) => res.blob())
+                    .then((b) => {
```

Comment:
what does 'b' stand for?

Reply:
its the blob, The overall code is pretty bad, i mentioned in the description that this needs a whole refactoring which includes moving the code to a utility, although i still don't know if this would fix the problem, thats why i want to test it out in the mobile/safari before investing any time in refactoring it! I still don't have a solution to test it in mobile without deploying to dev.

Reply:
ah, gotcha. Lets add a task for said refactor then to make sure we don't lose track of it